repo/content/committed_read_manager.go

```diff
@@ -115,6 +117,75 @@ type SharedManager struct {
 	internalLogger     *zap.SugaredLogger // backing logger for 'sharedBaseLogger'
 }
 
+func (sm *SharedManager) loadIndexBlob(ctx context.Context, ibi IndexBlobInfo, d gather.WriteBuffer) ([]Info, error) {
+	err := sm.st.GetBlob(ctx, ibi.BlobID, 0, -1, &d)
+	if err != nil {
+		return nil, err
+	}
+	q, err := ParseIndexBlob(ibi.BlobID, d.Bytes(), sm.format)
+	if err != nil {
+		return nil, err
+	}
+	return q, nil
+}
+
+func assign(iif Info, i int, m map[ID][2]Info) {
+	v := m[iif.GetContentID()]
+	v[i] = iif
+	m[iif.GetContentID()] = v
+}
+
+func (sm *SharedManager) ValidateIndexes(ctx context.Context) ([]string, error) {
+
+	d := gather.WriteBuffer{}
+	indexEntries := map[ID][2]Info{}
+
+	ibis0, _, err := sm.indexBlobManagerV0.listActiveIndexBlobs(ctx)
```

Comment:
nit: can we rename these (`ibis0, `ibi0`, `iifs`, etc.) to be more descriptive? Say, `activeIndexBlobsV0` or `indexBlobsV0`?